internal/gatewayapi/sort.go

```diff
@@ -85,8 +85,10 @@ func (x XdsIRRoutes) Less(i, j int) bool {
 func sortXdsIRMap(xdsIR resource.XdsIRMap) {
 	for _, irItem := range xdsIR {
 		for _, http := range irItem.HTTP {
-			// descending order
-			sort.Sort(sort.Reverse(XdsIRRoutes(http.Routes)))
+			if !http.PreserveRouteOrder {
```

Comment:
should skip the entire sort at the top level instead of inside the sort
https://github.com/envoyproxy/gateway/blob/731d0d0722532b654b9d508fea641e37ce63f98d/internal/gatewayapi/translator.go#L228

Reply:
the current approach is the right approach, because there may be an EP per GW, and so listeners within the entire XdsIRMap may have different values